### ISSUE TITLE
Make `id` a required prop for the TextField component

### ISSUE THREAD

**Issue by darrenvong** [Couchers-org/couchers#590]
According to the [accessibility section in the docs for `TextField`](https://material-ui.com/components/text-fields/#accessibility), adding an `id` is needed for Material UI to label our text fields correctly, which is important for keeping our app accessible for screen readers.

It's quite a low hanging fruit considering the library does the hard work of doing the rest right so we should fix this some time. A good first issue for anyone looking to get started!

**Comment by javiwq** [comment]
Is it alright if I try to implement this as my first contribution?

**Comment by Tay0108** [comment]
Done in https://github.com/Couchers-org/couchers/pull/693

**Comment by Tay0108** [comment]
Actually it is right now optional prop since we are using material-ui type for TextField, didn't notice it at first.

**Comment by lucaslcode** [comment]
Oops, neither did I. Easy fix, just create an interface that extends TextFieldProps and add id to it.

**Comment by javiwq** [comment]
So similar to this implementation I originally had on my branch?

![Screenshot 2021-02-25 at 09 12 36](https://user-images.githubusercontent.com/35485667/109123042-af29c400-7749-11eb-8192-d7a9547cf164.png)

I am not sure what went wrong in the merging - sorry about that! I can implement the fix today

**Comment by Tay0108** [comment]
I deleted it since we are using TextFieldProps from material UI and we are not omitting variant. And property id already exists there, but it's optional which I did not notice. Not sure if you can extend TextFieldProps tho.

**Comment by javiwq** [comment]
Ah, I see now, thanks! I'm also not sure if you can extend TextFieldProps, but I think we should be able to extend StandardTextFieldProps in this way?

**Comment by Tay0108** [comment]
But we cannot change variant then AFAIK

**Comment by lucaslcode** [comment]
I think you should be able to extend TextFieldProps?

**Comment by darrenvong** [comment]
Yeah you should be able to extend it, I did something almost identical in the Alert component to enforce `severity` as a required prop

**Comment by javiwq** [comment]
I found that an interface cannot extend TextFieldProps because [it is defined as:](https://unpkg.com/@material-ui/core@3.6.1/es/TextField/TextField.d.ts)
`type TextFieldProps = StandardTextFieldProps | FilledTextFieldProps | OutlinedTextFieldProps `
and an interface can only extend an object type with statically known members. 

I tried to implement via intersecting types but it still did not require the id. In the end I found a solution using [BaseTextFieldProps](https://unpkg.com/@material-ui/core@3.6.1/es/TextField/TextField.d.ts), which I could make work by also including the onChange? prop from TextFieldProps. 
I opened a PR with the changes, but I mention my thinking in case there's a more elegant solution I'm not seeing.
